helix-core/src/test/java/org/apache/helix/messaging/handling/TestHelixTaskExecutor.java

```diff
@@ -212,18 +213,33 @@ public TestStateTransitionHandlerFactory(String msgType, long delay) {
       _delay = delay;
     }
 
-    class TestStateTransitionMessageHandler extends MessageHandler {
+    class TestStateTransitionMessageHandler extends HelixStateTransitionHandler {
+      boolean _testIsMessageStaled;
+
       public TestStateTransitionMessageHandler(Message message, NotificationContext context) {
-        super(message, context);
+        super(null, null, message, context, null);
+        _testIsMessageStaled = false;
+      }
+
+      public TestStateTransitionMessageHandler(Message message, NotificationContext context,
+          CurrentState currentStateDelta) {
+        super(null, null, message, context, currentStateDelta);
+        _testIsMessageStaled = true;
+
+
       }
 
       @Override
-      public HelixTaskResult handleMessage() throws InterruptedException {
+      public HelixTaskResult handleMessage() {
         HelixTaskResult result = new HelixTaskResult();
         _processedMsgIds.put(_message.getMsgId(), _message.getMsgId());
         if (_delay > 0) {
           System.out.println("Sleeping..." + _delay);
-          Thread.sleep(_delay);
+          try{
+            Thread.sleep(_delay);
+          } catch (Exception e) {
+            assert (false);
```

Comment:
Can you help understand why we need to assert instead of throwing an `InterruptedException`?

Reply:
I think the reason is that the base handleMessage does not throw exception. The try-catch block here is only for handle exceptions in test.

Reply:
I don't think we need this change, as throwing an `InterruptedException` is good enough and cleaner to fail the test, unless the upstream caller catches and swallows the `InterruptedException`

Reply:
Please correct me if I am wrong. 
For the `InterruptedException`, we need to either catch it or add a declaration in method signature. Since the HelixStateTransitionHandler.handleMessage() does not throw exception, we could only do a try-catch here.